runtime/Java/src/org/antlr/v4/runtime/Parser.java

```diff
@@ -558,7 +558,7 @@ public void notifyErrorListeners(Token offendingToken, String msg,
 		charPositionInLine = offendingToken.getCharPositionInLine();
 
 		ANTLRErrorListener listener = getErrorListenerDispatch();
-		listener.syntaxError(this, offendingToken, line, charPositionInLine, msg, e);
+		listener.syntaxError(this, offendingToken, line, charPositionInLine, charPositionInLine + offendingToken.getText().length(), msg, e);
```

Comment:
:exclamation: The [`getStopIndex()`](http://www.antlr.org/api/Java/org/antlr/v4/runtime/Token.html#getStopIndex()) and `getStartIndex()` methods likely provide better behavior here, considering users may explicitly set the text. The following expression would be a more accurate representation of this change:

```
charPositionInLine + offendingToken.getStopIndex() - offendingToken.getStartIndex() + 1
```

:exclamation: This line assumes `offendingToken` does not span multiple lines, but in practice many tokens do.